README.md

```diff
@@ -0,0 +1,412 @@
+When the block ends, the indent returns to the previous indent level.
+If line-wrapping occurs, an extra indentation of +8 spaces should be added.
+
+### Vertical Whitespace
+
+There must be one blank line between function or method declarations.
+
+There must also be one blank line before and after access specifiers (`public`, `private` and `protected`) within a class definition.
+
+Within a method or function declaration, there should be blank lines separating different functionalities.
+It is convenient to start each functionality block with a comment.
+
+### Horizontal Whitespace
+
+There should not be any trailing whitespace at the end of the line.
+
+#### General
+
+When declaring variables, there must be only one space between the type and the variable name.
+If several variables are declared in the same line, there must be a space after the comma, but not before.
+This also applies to function declarations and callings.
+
+There must be no space before semicolons.
+
+When declaring arrays, the opening square bracket `[` should be always adjacent to the variable name.
+
+Before a backslash-newline `\` there should be a space.
+
+In `case` statements, there should not be a space before the colon `:`.
+
+#### Loops and Conditionals
+
+There must be a space between the keyword (`if`, `for`, `switch`, `while`, etc.) and the open parenthesis `(`.
+There should not be a space after the closing parenthesis `)` of a control statement.
+
+In `for` loops, there is always one space after each semicolon.
+
+#### Operators
+
+Assignment operators always have spaces around them.
+This also affects to compound assignment operators.
+
+Other binary arithmetic operators must also have spaces around them.
+This does not apply to the sign and the increment `++` and decrement `--`, which should be adjacent to the variable it affects.
+
+```c++
+v = -5 * x + y / --z;
+
+x += y++;
+```
+
+Logical operators `&&` and `||` should also have spaces around them.
+This does not apply to unary logical operator `!` which should be written adjacent to the variable it affects.
+
+Similarly, bitwise operators should have spaces around them, except the unary complement (bit inversion) operator `~` which should also be written adjacent to the variable it affects.
+
+Comparison operators ought to have spaces around them too.
+
+```c++
+if (param1 == 0 && param1 > 3 || !param2)
+{
+}
+```
+
+There should be no space around member operators `.` and `->`.
+
+Address-of `&` and de-reference `*` unary operators should be always adjacent to the variable they affect.
+
+When overloading an operator, there should be a space between the `operator` keyword and the operator being overloaded.
+
+```c++
+Subscriber& Subscriber::operator <<(
+        const qos::SubscriberQos& qos)
+{
+    this->qos(qos);
+    return *this;
+}
+```
+
+#### Templates
+
+There should be a space between closing angles `>`.
+
+```c++
+std::map<std::string, std::vector<DataWriterImpl*> > writers_;
+```
```

Comment:
I think this is the ugliest thing anybody can write. Eases confusion with the `>` operator, and doesn't clarify anything.

Reply:
Writing both angle brackets together (i.e. without a space in the middle), could also cause confusion with the `>>` operator, and is illegal on some old compilers. That said, support for `A<B<param>>` (without space) was added to the standard for C++11.

I see that we don't have a policy regarding closing parenthesis, so I think we should decide on both.

Should it be `my_function(my_adapter(param))` or `my_function(my_adapter(param) )` ?

My vote would go towards not using the space, being consistent with `my_function(param)`

Reply:
We force at least C++11 compliance, so the lack of the space between the angle brackets should not be a problem.
Hence, my vote goes to not using the space as well.

Reply:
This rule has been modified. It also includes the rule about nested parentheses. uncrustify.cfg file updated accordingly.